utils/src/chunked_polynomial.rs

```diff
@@ -0,0 +1,164 @@
+use ark_ff::{Field, Zero};
+use ark_poly::polynomial::{univariate::DensePolynomial, Polynomial, UVPolynomial};
+
+use rayon::prelude::*;
+
+#[derive(Clone)]
+pub struct ChunkedPolynomials<P> {
+    pub polys: Vec<P>,
+    pub degree: usize,
+}
+
+impl<P> Default for ChunkedPolynomials<P> {
+    fn default() -> ChunkedPolynomials<P> {
+        ChunkedPolynomials {
+            polys: vec![],
+            degree: 0,
+        }
+    }
+}
+
+impl<P> ChunkedPolynomials<P> {
+    pub fn add_chunk(&mut self, p: P) {
+        self.polys.push(p)
+    }
+}
+
+impl<F: Field> ChunkedPolynomials<DensePolynomial<F>> {
```

Comment:
I'm not a fan of this generic parameter. I think it's make more sense to define `ChunkedPolynomial` as

```rust
pub struct ChunkedPolynomial<F> {
    pub polys: Vec<DensePolynomial<F>>,
    pub degree: usize,
}
```

as this is what we're going to use it with mostly.

Reply:
> I'm not a fan of this generic parameter. I think it's make more sense to define `ChunkedPolynomial` as
> 
> ```rust
> pub struct ChunkedPolynomial<F> {
>     pub polys: Vec<DensePolynomial<F>>,
>     pub degree: usize,
> }
> ```
> 
> as this is what we're going to use it with mostly.

I thought about this before, the reason I use more generic parameters is to be compatible with the definition of `ScaledChunkedPolynomials`. Perhaps, we could refactor it.